File: library/src/test/java/de/westnordost/osmfeatures/CollectionUtilsTest.java
package de.westnordost.osmfeatures;

import org.junit.Test;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;
import java.util.Objects;

import static de.westnordost.osmfeatures.MapEntry.mapOf;
import static de.westnordost.osmfeatures.MapEntry.tag;
import static de.westnordost.osmfeatures.TestUtils.listOf;
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertNull;
import static org.junit.Assert.assertTrue;

public class CollectionUtilsTest
{

    @Test public void mapContainsEntry()
    {
        Map.Entry<String, String> ab = mapOf(tag("a", "b")).entrySet().iterator().next();
        Map.Entry<String, String> cd = mapOf(tag("c", "d")).entrySet().iterator().next();
        Map.Entry<String, String> ef = mapOf(tag("e", "f")).entrySet().iterator().next();
        Map<String, String> map = mapOf(tag("a", "b"), tag("c", "d"));

        assertTrue(CollectionUtils.mapContainsEntry(map, ab));
        assertTrue(CollectionUtils.mapContainsEntry(map, cd));
        assertFalse(CollectionUtils.mapContainsEntry(map, ef));
    }

    @Test public void numberOfContainedEntriesInMap()
    {
        Map<String, String> ab = mapOf(tag("a", "b"));
        Map<String, String> abcdef = mapOf(tag("a", "b"), tag("c", "d"), tag("e", "f"));

        Map<String, String> map = mapOf(tag("a", "b"), tag("c", "d"));

        assertEquals(0, CollectionUtils.numberOfContainedEntriesInMap(map, mapOf().entrySet()));
        assertEquals(1, CollectionUtils.numberOfContainedEntriesInMap(map, ab.entrySet()));
        assertEquals(2, CollectionUtils.numberOfContainedEntriesInMap(map, map.entrySet()));
        assertEquals(2, CollectionUtils.numberOfContainedEntriesInMap(map, abcdef.entrySet()));
    }

    @Test public void mapContainsAllEntries()
    {
        Map<String, String> ab = mapOf(tag("a", "b"));
        Map<String, String> abcdef = mapOf(tag("a", "b"), tag("c", "d"), tag("e", "f"));

        Map<String, String> map = mapOf(tag("a", "b"), tag("c", "d"));

        assertTrue(CollectionUtils.mapContainsAllEntries(map, mapOf().entrySet()));
        assertTrue(CollectionUtils.mapContainsAllEntries(map, ab.entrySet()));
        assertTrue(CollectionUtils.mapContainsAllEntries(map, map.entrySet()));
        assertFalse(CollectionUtils.mapContainsAllEntries(map, abcdef.entrySet()));
    }
}

File: library/src/commonTest/kotlin/de.westnordost.osmfeatures/CollectionUtilsTest.kt
package de.westnordost.osmfeatures

import org.junit.Test
import org.junit.Assert.assertEquals
import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue

class CollectionUtilsTest {
    @Test
    fun mapContainsEntry() {
        val ab = mapOf("a" to "b").entries.iterator().next()
        val cd = mapOf("c" to "d").entries.iterator().next()
        val ef = mapOf("e" to "f").entries.iterator().next()
        val map = mapOf("a" to "b", "c" to "d")
        assertTrue(CollectionUtils.mapContainsEntry(map, ab))
        assertTrue(CollectionUtils.mapContainsEntry(map, cd))
        assertFalse(CollectionUtils.mapContainsEntry(map, ef))
    }

    @Test
    fun numberOfContainedEntriesInMap() {
        val ab = mapOf("a" to "b")
        val abcdef = mapOf("a" to "b", "c" to "d", "e" to "f")
        val map = mapOf("a" to "b", "c" to "d")
        assertEquals(0, CollectionUtils.numberOfContainedEntriesInMap(map, emptyMap<String, String>().entries))
        assertEquals(1, CollectionUtils.numberOfContainedEntriesInMap(map, ab.entries))
        assertEquals(2, CollectionUtils.numberOfContainedEntriesInMap(map, map.entries))
        assertEquals(2, CollectionUtils.numberOfContainedEntriesInMap(map, abcdef.entries))
    }

    @Test
    fun mapContainsAllEntries() {
        val ab = mapOf("a" to "b")
        val abcdef = mapOf("a" to "b", "c" to "d", "e" to "f")
        val map = mapOf("a" to "b", "c" to "d")
        assertTrue(CollectionUtils.mapContainsAllEntries(map, emptyMap<String, String>().entries))
        assertTrue(CollectionUtils.mapContainsAllEntries(map, ab.entries))
        assertTrue(CollectionUtils.mapContainsAllEntries(map, map.entries))
        assertFalse(CollectionUtils.mapContainsAllEntries(map, abcdef.entries))
    }
}
